lib/fluent/plugin/out_influxdb.rb

```diff
@@ -131,9 +135,30 @@ def write(chunk)
         :values    => values,
         :tags      => tags,
       }
-      points << point
+      retention_policy = nil
+      unless @retention_policy_key.nil?
+        retention_policy = record.delete(@retention_policy_key)
+        unless points.nil?
+          if !retention_policy.nil? && retention_policy != @default_retention_policy
+            # flush the retention policy first
+            @influxdb.write_points(points, nil, @default_retention_policy)
```

Comment:
When a record accompanied by `retention_policy_key` occurs for the first time, the preceding points should be written with the default retention policy.  So the part in question is correct.

BTW, in the line 138, `retention_policy` should have been initialized to `@default_retention_policy` instead of `nil`.  I'm going to fix this accordingly